test-d/destructurable.ts

```diff
@@ -0,0 +1,50 @@
+import {expectType} from 'tsd';
+import {type Destructurable} from '../index';
+
+// Typological use cases
+type Success = {
+	type: 'success';
+	value: string;
+};
+type Failure = {
+	type: 'failure';
+	error: Error;
+};
+type Result = Destructurable<Success | Failure>;
+function doSomething(): Result {
+	return Math.random() > 0.5
```

Comment:
Do you really want tests which give different results, based on a random value?  This would lead potentially to some tests passing and others failing, and no, one couldn't reproduce a failure consistently.

Reply:
Thank you for your comment, ajvincent. I appreciate your concern about the use of a random value in tests, and how it could potentially be misleading.

While the tests in type-fest are type-level checks and the random value doesn't actually impact the test results, I understand that it could lead to confusion. In light of your feedback, I have already updated the test cases to remove the random value and opted for deterministic code. This way, we can ensure clarity and consistency for anyone reviewing the tests in the future.